internal/db/datatypes.go

```diff
@@ -0,0 +1,160 @@
+package db
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"net/url"
+	"regexp"
+)
+
+var (
+	ErrValue = fmt.Errorf("%w: invalid value for type", ErrData)
+)
+
+type Datatype struct {
+	ID            uuid.UUID
+	Name          string
+	Validator     Code
+	StorageFormat StorageFormat
+}
+
+type StorageFormat int64
+
+const (
+	BoolFormat StorageFormat = iota
+	IntFormat
+	StringFormat
+	FloatFormat
+	UUIDFormat
+)
+
+var storageFormat map[StorageFormat]interface{} = map[StorageFormat]interface{}{
+	BoolFormat:   false,
+	IntFormat:    int64(0),
+	StringFormat: "",
+	FloatFormat:  0.0,
+	UUIDFormat:   uuid.UUID{},
+}
+
+var nativeDatatypes map[uuid.UUID]Datatype = map[uuid.UUID]Datatype{
+	Bool.ID:         Bool,
+	Int.ID:          Int,
+	Enum.ID:         Enum,
+	String.ID:       String,
+	Text.ID:         Text,
+	EmailAddress.ID: EmailAddress,
+	UUID.ID:         UUID,
+	Float.ID:        Float,
+	URL.ID:          URL,
+}
+
+func boolValidatorFunc(value interface{}) (interface{}, error) {
```

Comment:
I'd like to see all of these validator funcs at least pulled into a different file/subpackage like "datatypes/core.go"

Reply:
Just the functions, or also the meta model structs as well?

Reply:
Given the refactor, I think it might make sense to put these in the new package. Thoughts?